scripts/test.sh

```diff
@@ -340,24 +340,14 @@ function shellcheck_pass {
 }
 
 function shellws_pass {
-  TAB=$'\t'
   log_callout "Ensuring no tab-based indention in shell scripts"
   local files
-  files=$(find ./ -name '*.sh' -print0 | xargs -0 )
-  # shellcheck disable=SC2206
-  files=( ${files[@]} "./scripts/build-binary.sh" "./scripts/build-docker.sh" "./scripts/release.sh" )
-  log_cmd "grep -E -n $'^ *${TAB}' ${files[*]}"
-  # shellcheck disable=SC2086
-  if grep -E -n $'^ *${TAB}' "${files[@]}" | sed $'s|${TAB}|[\\\\tab]|g'; then
+  if files=$(find . -name '*.sh' -print0 | xargs -0 grep -E -n $'^\s*\t'); then
     log_error "FAIL: found tab-based indention in bash scripts. Use '  ' (double space)."
-    local files_with_tabs
-    files_with_tabs=$(grep -E -l $'^ *\\t' "${files[@]}")
-    log_warning "Try: sed -i 's|\\t|  |g' $files_with_tabs"
-    return 1
-  else
-    log_success "SUCCESS: no tabulators found."
-    return 0
+    log_warning "Try: sed -i 's|\\t|  |g' $(echo "${files}" | cut -d: -f1 | paste -sd ' ')"
```

Comment:
It has two minor issues:
- It may print duplicate files name if there are TABs in multiple lines in one file. For example,
  ```
  Try: sed -i 's|\t|  |g' ./scripts/release.sh ./scripts/release.sh
  ```
- It doesn't work on Mac OS, it should be caused by `paste`'s different behaviour.  Replacing `paste -sd ' '` with `tr '\n' ' '` can resolve this.

Reply:
Good point. I'll always make sure to test the scripts in MacOS to ensure BSD compatibility.

However, running `make fix` also fixes this issue. IMHO, it would be more valuable to show the files and the lines with the error as output and suggest running `make fix`. I aimed to keep the same behavior, but I think this would be better.

Reply:
i.e.,

```
'shellws' started at Fri Dec 13 11:52:19 AM PST 2024
Ensuring no tab-based indention in shell scripts
FAIL: found tab-based indention in the following bash scripts. Use '  ' (double space):
./scripts/test.sh:345:          if files=$(find . -name '*.sh' -print0 | xargs -0 grep -E -n $'^\s*\t'); then
./scripts/test.sh:351:          log_success "SUCCESS: no tabulators found."
Run make fix to address the issue.
FAIL: 'shellws' FAILED at Fri Dec 13 11:52:19 AM PST 2024
```

Reply:
> it would be more valuable to show the files and the lines with the error as output and suggest running `make fix`

Sounds good to me.

Reply:
Done.